lib/ui/painting/codec.cc

```diff
@@ -376,12 +413,14 @@ MultiFrameCodec::MultiFrameCodec(std::unique_ptr<SkCodec> codec,
   for (size_t frameIndex = 0; frameIndex < frameInfos_.size(); frameIndex++) {
     const auto& frameInfo = frameInfos_[frameIndex];
     if (frameInfo.fRequiredFrame != SkCodec::kNoFrame) {
-      frameBitmaps_[frameInfo.fRequiredFrame] =
-          std::make_unique<DecodedFrame>(/*required=*/true);
+      if (frameBitmaps_[frameInfo.fRequiredFrame] == nullptr) {
```

Comment:
Wouldn't we have always already set the frameBitmaps_ entry for the required frame (in line 423)

Reply:
I think so, but wasn't totally sure. I was putting this check here in case for some reason there's a looping image and frame 0 depends on the very last frame. It seems unlikely but I couldn't find anything to rule it out. Do you know if they're always sequential or not?